src/components/DataTable/DataTable.jsx

```diff
@@ -0,0 +1,260 @@
+		event.bubbles = false;
+		onSort(field, {props: this.props, event});
+	},
+
+	render() {
+
+		const {
+			className,
+			data,
+			hasDetails,
+			isSelectable,
+			...passThroughs
+		} = this.props;
+
+
+		const childComponentElements = findAllChildComponents(this.props, [DataTable.Column, DataTable.ColumnGroup]);
+		const flattenedColumns = _.reduce(childComponentElements, (acc, childComponentElement) => {
+			if (childComponentElement.type === DataTable.Column) {
+				return acc.concat([{props: childComponentElement.props, columnGroupProps: null}]);
+			}
+			if (childComponentElement.type === DataTable.ColumnGroup) {
+				return acc.concat(_.map(findAllChildComponents(childComponentElement.props, [DataTable.Column]), (columnChildComponent) => ({ props: columnChildComponent.props, columnGroupProps: childComponentElement.props })));
+			}
+		}, []);
+
+		const hasGroupedColumns = _.some(childComponentElements, (childComponentElement) => childComponentElement.type === DataTable.ColumnGroup);
+		return (
+			<ScrollTable {...passThroughs} className={boundClassNames('&', className)}>
+				<Thead>
+					<Tr>
+						{isSelectable ? (
+							<Th hasCheckbox rowSpan={hasGroupedColumns ? 2 : null}><Checkbox isSelected={_.every(data, 'isSelected')} onSelect={this.handleSelectAll} /></Th>
+						) : null}
+						{_.map(childComponentElements, ({ props, type }, index) => type === DataTable.Column ? (
+							<Th
+								{..._.omit(props, ['field', 'children', 'width', 'title'])}
+								onClick={DataTable.shouldColumnHandleSort(props) ? _.partial(this.handleSort, props.field) : null}
+								style={{
+									width: props.width
+								}}
+								rowSpan={hasGroupedColumns ? 2 : null}
+								key={_.get(props, 'field', index)}
+							>
+								{props.title || props.children}
+							</Th>
+						) : (
+							<Th
+								colSpan={_.size(DataTable.Column.findInChildren(props.children))}
+								{..._.omit(props, ['field', 'children', 'width', 'title'])}
+								key={_.get(props, 'field', index)}
+							>
+								{props.title || props.children}
+							</Th>
+						))}
+					</Tr>
+					{hasGroupedColumns ? (
+						<Tr>
+							{_.reduce(flattenedColumns, (acc, { props: columnProps, columnGroupProps }, index) => acc.concat(_.isNull(columnGroupProps) ? [] : [(
+								<Th
+									{..._.omit(columnProps, ['field', 'children', 'width', 'title'])}
+									onClick={DataTable.shouldColumnHandleSort(columnProps) ? _.partial(this.handleSort, columnProps.field) : null}
+									style={{
+										width: columnProps.width
+									}}
+									key={_.get(columnProps, 'field', index)}
+								>
+									{columnProps.title || columnProps.children}
+								</Th>
+							)]), [])}
+						</Tr>
+					) : null}
+				</Thead>
+				<Tbody>
+					{_.map(data, (row, index) => (
+						<Tr isSelected={row.isSelected} isActive={row.isActive} onClick={_.partial(this.handleRowClick, index)} hasDetails={hasDetails} key={'row' + index}>
```

Comment:
These long lines are hard to read. Again, it's okay for now but let's drop them in the future:

```
<Tr
	isSelected={row.isSelected}
	isActive={row.isActive}
	onClick={_.partial(this.handleRowClick, index)}
	hasDetails={hasDetails}
	key={'row' + index}
>
```